libheif/heif_emscripten.h

```diff
@@ -112,7 +112,7 @@ static emscripten::val heif_js_decode_image(struct heif_image_handle* handle,
   result.set("width", width);
   int height = heif_image_get_height(image, heif_channel_Y);
   result.set("height", height);
-  std::string data;
+  std::basic_string<unsigned char> data;
```

Comment:
This was needed because Embind was messing up the data transfer when using a normal std::string, however, the data transfer between strings looks very odd to me right now because I think what currently happens is:
- WASM sends pointer to string to JS
- JS reads binary data from pointer position
- JS converts binary data into string
- libheif converts string back to binary data (`post.js`)

I believe we can remove the string conversion, probably also speeds things up a bit. I'm currently talking to Emscripten what the best way to do that is in Embind.

Reply:
Converning passing the data around: do you think [this (answer item 3)](https://stackoverflow.com/questions/53602955/using-emscripten-how-to-get-c-uint8-t-array-to-js-blob-or-uint8array) might work?

Reply:
Yes! `result.set("data", emscripten::val(emscripten::typed_memory_view(data.size(), data.data())));` seems to work perfectly! It will give you a `Uint8Array` directly in JS, however, I'm not completely sure who then becomes owner of the data buffer and who has to free it, would have to be looked into to prevent memory leaks.